Record parsing deadlock

**Detailed Description**
Importer record parsing thread stopped processing files in performance environment. There were no active queries or locks in the database. A restart of the importer allowed it to recover.

**Actual Behavior**
```
Apr 24 09:44:39 mirrornode-performance-parser-1 java[30055]: 2021-04-24T03:44:39.530-0600 ERROR scheduling-5 c.h.m.i.p.r.RecordFileParser Error parsing file 2021-04-24T09_36_52.014157000Z.rcd com.hedera.mirror.importer.exception.ParserException: Error copying 18241 items to table topic_message\\n        at 
com.hedera.mirror.importer.parser.PgCopy.copy(PgCopy.java:93)\\n        at 
com.hedera.mirror.importer.parser.record.entity.sql.SqlEntityListener.executeBatches(SqlEntityListener.java:198)\\n        at com.hedera.mirror.importer.parser.record.entity.sql.SqlEntityListener.onEnd(SqlEntityListener.java:160)\\n        at com.hedera.mirror.importer.parser.record.entity.sql.SqlEntityListener.onEnd(SqlEntityListener.java:69)\\n        at com.hedera.mirror.importer.parser.record.RecordFileParser.parse(RecordFileParser.java:165)\\n        at com.hedera.mirror.importer.parser.record.RecordFileParser$$FastClassBySpringCGLIB$$8b9310db.invoke(<generated>)\\n        at 
org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)\\n        at 
org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:779)\\n        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)\\n        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)\\n        at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)\\n        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388)\\n        at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)\\n        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)\\n        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)\\n        at org.springframework.retry.interceptor.RetryOperationsInterceptor$1.doWithRetry(RetryOperationsInterceptor.java:93)\\n        at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:329)\\n        at 
org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:209)\\n        at org.springframework.retry.interceptor.RetryOperationsInterceptor.invoke(RetryOperationsInterceptor.java:119)\\n        at org.springframework.retry.annotation.AnnotationAwareRetryOperationsInterceptor.invoke(AnnotationAwareRetryOperationsInterceptor.java:163)\\n        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)\\n        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)\\n        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:692)\\n        at com.hedera.mirror.importer.parser.record.RecordFileParser$$EnhancerBySpringCGLIB$$d58df86e.parse(<generated>)\\n        at 
jdk.internal.reflect.GeneratedMethodAccessor137.invoke(Unknown Source)\\n        at 
java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\\n        at 
java.base/java.lang.reflect.Method.invoke(Method.java:566)\\n        at 
org.springframework.messaging.handler.invocation.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:171)\\n        at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.invoke(InvocableHandlerMethod.java:120)\\n        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper$HandlerMethod.invoke(MessagingMethodInvokerHelper.java:1105)\\n        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper.invokeHandlerMethod(MessagingMethodInvokerHelper.java:583)\\n        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper.processInternal(MessagingMethodInvokerHelper.java:478)\\n        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper.process(MessagingMethodInvokerHelper.java:356)\\n        at org.springframework.integration.handler.MethodInvokingMessageProcessor.processMessage(MethodInvokingMessageProcessor.java:108)\\n        at org.springframework.integration.handler.ServiceActivatingHandler.handleRequestMessage(ServiceActivatingHandler.java:105)\\n        at org.springframework.integration.handler.AbstractReplyProducingMessageHandler.handleMessageInternal(AbstractReplyProducingMessageHandler.java:134)\\n        at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:56)\\n        at org.springframework.integration.endpoint.PollingConsumer.handleMessage(PollingConsumer.java:158)\\n        at org.springframework.integration.endpoint.AbstractPollingEndpoint.messageReceived(AbstractPollingEndpoint.java:449)\\n        at org.springframework.integration.endpoint.AbstractPollingEndpoint.doPoll(AbstractPollingEndpoint.java:435)\\n        at org.springframework.integration.endpoint.AbstractPollingEndpoint.pollForMessage(AbstractPollingEndpoint.java:387)\\n        at org.springframework.integration.endpoint.AbstractPollingEndpoint.lambda$null$4(AbstractPollingEndpoint.java:331)\\n        at org.springframework.integration.util.ErrorHandlingTaskExecutor.lambda$execute$0(ErrorHandlingTaskExecutor.java:57)\\n        at org.springframework.core.task.SyncTaskExecutor.execute(SyncTaskExecutor.java:50)\\n        at org.springframework.integration.util.ErrorHandlingTaskExecutor.execute(ErrorHandlingTaskExecutor.java:55)\\n        at org.springframework.integration.endpoint.AbstractPollingEndpoint.lambda$createPoller$5(AbstractPollingEndpoint.java:328)\\n        at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54)\\n        at org.springframework.scheduling.concurrent.ReschedulingRunnable.run(ReschedulingRunnable.java:95)\\n        at 
java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)\\n        at 
java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)\\n        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)\\n        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\\n        at 
java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\\n        at 
java.base/java.lang.Thread.run(Thread.java:834)\\nCaused by: java.io.IOException: Write to copy failed.\\n        at 
org.postgresql.copy.PGCopyOutputStream.write(PGCopyOutputStream.java:95)\\n        at 
com.fasterxml.jackson.dataformat.csv.impl.UTF8Writer.write(UTF8Writer.java:135)\\n        at 
com.fasterxml.jackson.dataformat.csv.impl.CsvEncoder._flushBuffer(CsvEncoder.java:1140)\\n        at 
com.fasterxml.jackson.dataformat.csv.impl.CsvEncoder.write(CsvEncoder.java:390)\\n        at 
com.fasterxml.jackson.dataformat.csv.CsvGenerator.writeNumber(CsvGenerator.java:811)\\n        at com.fasterxml.jackson.databind.ser.std.NumberSerializers$IntegerSerializer.serialize(NumberSerializers.java:152)\\n        at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:728)\\n        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:755)\\n        at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:178)\\n        at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serializeContents(IndexedListSerializer.java:119)\\n        at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serialize(IndexedListSerializer.java:79)\\n        at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serialize(IndexedListSerializer.java:18)\\n        at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:480)\\n        at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:319)\\n        at com.fasterxml.jackson.databind.ObjectWriter$Prefetch.serialize(ObjectWriter.java:1516)\\n        at 
com.fasterxml.jackson.databind.ObjectWriter._writeValueAndClose(ObjectWriter.java:1217)\\n        at 
com.fasterxml.jackson.databind.ObjectWriter.writeValue(ObjectWriter.java:1043)\\n        at 
com.hedera.mirror.importer.parser.PgCopy.copy(PgCopy.java:87)\\n        ... 53 more\\n        
Suppressed: java.io.IOException: Write to copy failed.\\n                at 
org.postgresql.copy.PGCopyOutputStream.write(PGCopyOutputStream.java:95)\\n                at 
com.fasterxml.jackson.dataformat.csv.impl.UTF8Writer.write(UTF8Writer.java:135)\\n                at 
com.fasterxml.jackson.dataformat.csv.impl.CsvEncoder._flushBuffer(CsvEncoder.java:1140)\\n                at 
com.fasterxml.jackson.dataformat.csv.impl.CsvEncoder.close(CsvEncoder.java:990)\\n                at 
com.fasterxml.jackson.dataformat.csv.CsvGenerator.close(CsvGenerator.java:497)\\n                at 
com.fasterxml.jackson.databind.util.ClassUtil.closeOnFailAndThrowAsIOE(ClassUtil.java:486)\\n                at com.fasterxml.jackson.databind.ObjectWriter._writeValueAndClose(ObjectWriter.java:1219)\\n                ... 55 more\\n        
Caused by: org.postgresql.util.PSQLException: Database connection failed when writing to copy\\n                at 
org.postgresql.core.v3.QueryExecutorImpl.writeToCopy(QueryExecutorImpl.java:1050)\\n                at 
org.postgresql.core.v3.CopyInImpl.writeToCopy(CopyInImpl.java:37)\\n                at 
org.postgresql.copy.PGCopyOutputStream.writeToCopy(PGCopyOutputStream.java:135)\\n                at 
org.postgresql.copy.PGCopyOutputStream.write(PGCopyOutputStream.java:93)\\n                ... 61 more\\n        
Caused by: java.net.SocketException: Broken pipe (Write failed)\\n                at 
java.base/java.net.SocketOutputStream.socketWrite0(Native Method)\\n                at 
java.base/java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:110)\\n                at 
java.base/java.net.SocketOutputStream.write(SocketOutputStream.java:150)\\n                at 
java.base/sun.security.ssl.SSLSocketOutputRecord.deliver(SSLSocketOutputRecord.java:319)\\n                at 
java.base/sun.security.ssl.SSLSocketImpl$AppOutputStream.write(SSLSocketImpl.java:1202)\\n                at 
java.base/java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:81)\\n                at 
java.base/java.io.BufferedOutputStream.write(BufferedOutputStream.java:122)\\n                at 
org.postgresql.core.PGStream.send(PGStream.java:389)\\n                at 
org.postgresql.core.v3.QueryExecutorImpl.writeToCopy(QueryExecutorImpl.java:1048)\\n                ... 64 more\\n        
Suppressed: java.io.IOException: Ending write to copy failed.\\n                at 
org.postgresql.copy.PGCopyOutputStream.close(PGCopyOutputStream.java:115)\\n                at 
com.hedera.mirror.importer.parser.PgCopy.copy(PgCopy.java:86)\\n                ... 53 more\\n        
Caused by: org.postgresql.util.PSQLException: Database connection failed when writing to copy\\n                at 
org.postgresql.core.v3.QueryExecutorImpl.writeToCopy(QueryExecutorImpl.java:1050)\\n                at 
org.postgresql.core.v3.CopyInImpl.writeToCopy(CopyInImpl.java:37)\\n                at 
org.postgresql.copy.PGCopyOutputStream.endCopy(PGCopyOutputStream.java:176)\\n                at 
org.postgresql.copy.PGCopyOutputStream.close(PGCopyOutputStream.java:113)\\n                ... 54 more\\n        
Caused by: java.net.SocketException: Broken pipe (Write failed)\\n                at 
java.base/java.net.SocketOutputStream.socketWrite0(Native Method)\\n                at 
java.base/java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:110)\\n                at 
java.base/java.net.SocketOutputStream.write(SocketOutputStream.java:150)\\n                at 
java.base/sun.security.ssl.SSLSocketOutputRecord.deliver(SSLSocketOutputRecord.java:319)\\n                at 
java.base/sun.security.ssl.SSLSocketImpl$AppOutputStream.write(SSLSocketImpl.java:1202)\\n                at 
java.base/java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:81)\\n                at 
java.base/java.io.BufferedOutputStream.write(BufferedOutputStream.java:122)\\n                at 
org.postgresql.core.PGStream.send(PGStream.java:389)\\n                at 
org.postgresql.core.v3.QueryExecutorImpl.writeToCopy(QueryExecutorImpl.java:1048)\\n                ... 57 more\\n
Caused by: org.postgresql.util.PSQLException: Database connection failed when writing to copy\\n        at 
org.postgresql.core.v3.QueryExecutorImpl.writeToCopy(QueryExecutorImpl.java:1050)\\n        at 
org.postgresql.core.v3.CopyInImpl.writeToCopy(CopyInImpl.java:37)\\n        at 
org.postgresql.copy.PGCopyOutputStream.writeToCopy(PGCopyOutputStream.java:135)\\n        at 
org.postgresql.copy.PGCopyOutputStream.write(PGCopyOutputStream.java:93)\\n        ... 70 more\\n
Caused by: java.net.SocketException: Connection reset by peer (Write failed)\\n        at 
java.base/java.net.SocketOutputStream.socketWrite0(Native Method)\\n        at 
java.base/java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:110)\\n        at 
java.base/java.net.SocketOutputStream.write(SocketOutputStream.java:150)\\n        at 
java.base/sun.security.ssl.SSLSocketOutputRecord.deliver(SSLSocketOutputRecord.java:319)\\n        at 
java.base/sun.security.ssl.SSLSocketImpl$AppOutputStream.write(SSLSocketImpl.java:1202)\\n        at 
java.base/java.io.BufferedOutputStream.write(BufferedOutputStream.java:123)\\n        at 
org.postgresql.core.PGStream.send(PGStream.java:389)\\n        at org.postgresql.core.v3.QueryExecutorImpl.writeToCopy(QueryExecutorImpl.java:1048)\\n        ... 73 more
```

Stacktrace
```
"scheduling-4" #39 prio=5 os_prio=0 cpu=176736.39ms elapsed=49334.78s tid=0x00007f2b4584f800 nid=0x761e waiting on condition  [0x00007f2abc149000]
   java.lang.Thread.State: WAITING (parking)
        at jdk.internal.misc.Unsafe.park(java.base@11.0.10/Native Method)
        - parking to wait for  <0x00000005828f3280> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.park(java.base@11.0.10/LockSupport.java:194)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(java.base@11.0.10/AbstractQueuedSynchronizer.java:2081)
        at java.util.concurrent.LinkedBlockingQueue.put(java.base@11.0.10/LinkedBlockingQueue.java:341)
        at org.springframework.integration.channel.QueueChannel.doSend(QueueChannel.java:120)
        at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:317)
        at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:272)
        at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:187)
        at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:166)
        at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:47)
        at org.springframework.messaging.core.AbstractMessageSendingTemplate.send(AbstractMessageSendingTemplate.java:109)
        at org.springframework.integration.router.AbstractMessageRouter.doSend(AbstractMessageRouter.java:213)
        at org.springframework.integration.router.AbstractMessageRouter.handleMessageInternal(AbstractMessageRouter.java:195)
        at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:56)
        at org.springframework.integration.dispatcher.AbstractDispatcher.tryOptimizedDispatch(AbstractDispatcher.java:115)
        at org.springframework.integration.dispatcher.UnicastingDispatcher.doDispatch(UnicastingDispatcher.java:133)
        at org.springframework.integration.dispatcher.UnicastingDispatcher.dispatch(UnicastingDispatcher.java:106)
        at org.springframework.integration.channel.AbstractSubscribableChannel.doSend(AbstractSubscribableChannel.java:72)
        at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:317)
        at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:272)
        at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:187)
        at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:166)
        at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:47)
        at org.springframework.messaging.core.AbstractMessageSendingTemplate.send(AbstractMessageSendingTemplate.java:109)
        at org.springframework.messaging.core.AbstractMessageSendingTemplate.convertAndSend(AbstractMessageSendingTemplate.java:151)
        at org.springframework.messaging.core.AbstractMessageSendingTemplate.convertAndSend(AbstractMessageSendingTemplate.java:143)
        at org.springframework.integration.gateway.MessagingGatewaySupport.send(MessagingGatewaySupport.java:428)
        at org.springframework.integration.gateway.GatewayProxyFactoryBean.sendOrSendAndReceive(GatewayProxyFactoryBean.java:662)
        at org.springframework.integration.gateway.GatewayProxyFactoryBean.invokeGatewayMethod(GatewayProxyFactoryBean.java:583)
        at org.springframework.integration.gateway.GatewayProxyFactoryBean.doInvoke(GatewayProxyFactoryBean.java:550)
        at org.springframework.integration.gateway.GatewayProxyFactoryBean.invoke(GatewayProxyFactoryBean.java:539)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:215)
        at com.sun.proxy.$Proxy215.verified(Unknown Source)
        at com.hedera.mirror.importer.downloader.Downloader.onVerified(Downloader.java:479)
        at com.hedera.mirror.importer.downloader.Downloader.verifySigsAndDownloadDataFiles(Downloader.java:441)
        at com.hedera.mirror.importer.downloader.Downloader.downloadNextBatch(Downloader.java:178)
        at com.hedera.mirror.importer.downloader.record.RecordFileDownloader.download(RecordFileDownloader.java:57)

"scheduling-5" #40 prio=5 os_prio=0 cpu=206197.40ms elapsed=49334.78s tid=0x00007f2b44716800 nid=0x761f in Object.wait()  [0x00007f2a93ffd000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(java.base@11.0.10/Native Method)
        - waiting on <no object reference available>
        at java.lang.Object.wait(java.base@11.0.10/Object.java:328)
        at org.postgresql.core.v3.QueryExecutorImpl.waitOnLock(QueryExecutorImpl.java:207)
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:289)
        - waiting to re-lock in wait() <0x0000000594200000> (a org.postgresql.core.v3.QueryExecutorImpl)
        at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:849)
        at org.postgresql.jdbc.PgConnection.rollback(PgConnection.java:892)
        at com.zaxxer.hikari.pool.ProxyConnection.rollback(ProxyConnection.java:396)
        at com.zaxxer.hikari.pool.HikariProxyConnection.rollback(HikariProxyConnection.java)
        at org.hibernate.resource.jdbc.internal.AbstractLogicalConnectionImplementor.rollback(AbstractLogicalConnectionImplementor.java:121)
        at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl$TransactionDriverControlImpl.rollback(JdbcResourceLocalTransactionCoordinatorImpl.java:304)
        at org.hibernate.engine.transaction.internal.TransactionImpl.rollback(TransactionImpl.java:142)
        at org.springframework.orm.jpa.JpaTransactionManager.doRollback(JpaTransactionManager.java:589)
        at org.springframework.transaction.support.AbstractPlatformTransactionManager.processRollback(AbstractPlatformTransactionManager.java:835)
        at org.springframework.transaction.support.AbstractPlatformTransactionManager.rollback(AbstractPlatformTransactionManager.java:809)
        at org.springframework.transaction.interceptor.TransactionAspectSupport.completeTransactionAfterThrowing(TransactionAspectSupport.java:672)
        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:392)
        at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)
        at org.springframework.retry.interceptor.RetryOperationsInterceptor$1.doWithRetry(RetryOperationsInterceptor.java:93)
        at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:329)
        at org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:209)
        at org.springframework.retry.interceptor.RetryOperationsInterceptor.invoke(RetryOperationsInterceptor.java:119)
        at org.springframework.retry.annotation.AnnotationAwareRetryOperationsInterceptor.invoke(AnnotationAwareRetryOperationsInterceptor.java:163)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:692)
        at com.hedera.mirror.importer.parser.record.RecordFileParser$$EnhancerBySpringCGLIB$$d58df86e.parse(<generated>)
```

Logs, stacktrace and metrics:
[record-thread-deadlock.tar.gz](https://github.com/hashgraph/hedera-mirror-node/files/6379743/record-thread-deadlock.tar.gz)

**Expected Behavior**
Files process successfully or rollback and retry if unsuccessful.

**Environment:**
- Java: AdoptOpenJDK 11.0.10
- OS: Ubuntu 18.04
- Version: 0.32.0-rc1

**Additional Context**

--- a/charts/hedera-mirror-common/Chart.yaml
+++ b/charts/hedera-mirror-common/Chart.yaml
@@ -1,5 +1,5 @@
 apiVersion: v2
-appVersion: "0.32.0-rc1"
+appVersion: "0.32.0-rc2"
 dependencies:
   - name: loki
     condition: loki.enabled
@@ -33,4 +33,4 @@ maintainers:
 name: hedera-mirror-common
 sources:
   - https://github.com/hashgraph/hedera-mirror-node
-version: 0.19.0-rc1
+version: 0.19.0-rc2

--- a/charts/hedera-mirror-grpc/Chart.yaml
+++ b/charts/hedera-mirror-grpc/Chart.yaml
@@ -1,5 +1,5 @@
 apiVersion: v2
-appVersion: "0.32.0-rc1"
+appVersion: "0.32.0-rc2"
 description: GRPC API for the Hedera Mirror Node
 home: https://github.com/hashgraph/hedera-mirror-node
 icon: https://camo.githubusercontent.com/cca6b767847bb8ca5c7059481ba13a5fc81c5938/68747470733a2f2f7777772e6865646572612e636f6d2f6c6f676f2d6361706974616c2d686261722d776f72646d61726b2e6a7067
@@ -11,4 +11,4 @@ maintainers:
 name: hedera-mirror-grpc
 sources:
   - https://github.com/hashgraph/hedera-mirror-node
-version: 0.19.0-rc1
+version: 0.19.0-rc2

--- a/charts/hedera-mirror-importer/Chart.yaml
+++ b/charts/hedera-mirror-importer/Chart.yaml
@@ -1,5 +1,5 @@
 apiVersion: v2
-appVersion: "0.32.0-rc1"
+appVersion: "0.32.0-rc2"
 description: Hedera Mirror Importer downloads and validates file streams from the cloud and imports them into the database
 home: https://github.com/hashgraph/hedera-mirror-node
 icon: https://camo.githubusercontent.com/cca6b767847bb8ca5c7059481ba13a5fc81c5938/68747470733a2f2f7777772e6865646572612e636f6d2f6c6f676f2d6361706974616c2d686261722d776f72646d61726b2e6a7067
@@ -11,4 +11,4 @@ maintainers:
 name: hedera-mirror-importer
 sources:
   - https://github.com/hashgraph/hedera-mirror-node
-version: 0.19.0-rc1
+version: 0.19.0-rc2

--- a/charts/hedera-mirror-monitor/Chart.yaml
+++ b/charts/hedera-mirror-monitor/Chart.yaml
@@ -1,5 +1,5 @@
 apiVersion: v2
-appVersion: "0.32.0-rc1"
+appVersion: "0.32.0-rc2"
 description: End to End Monitor for the Hedera Mirror Node
 home: https://github.com/hashgraph/hedera-mirror-node
 icon: https://camo.githubusercontent.com/cca6b767847bb8ca5c7059481ba13a5fc81c5938/68747470733a2f2f7777772e6865646572612e636f6d2f6c6f676f2d6361706974616c2d686261722d776f72646d61726b2e6a7067
@@ -11,4 +11,4 @@ maintainers:
 name: hedera-mirror-monitor
 sources:
   - https://github.com/hashgraph/hedera-mirror-node
-version: 0.19.0-rc1
+version: 0.19.0-rc2

--- a/charts/hedera-mirror-rest/Chart.yaml
+++ b/charts/hedera-mirror-rest/Chart.yaml
@@ -1,5 +1,5 @@
 apiVersion: v2
-appVersion: "0.32.0-rc1"
+appVersion: "0.32.0-rc2"
 description: REST API for the Hedera Mirror Node
 home: https://github.com/hashgraph/hedera-mirror-node
 icon: https://camo.githubusercontent.com/cca6b767847bb8ca5c7059481ba13a5fc81c5938/68747470733a2f2f7777772e6865646572612e636f6d2f6c6f676f2d6361706974616c2d686261722d776f72646d61726b2e6a7067
@@ -11,4 +11,4 @@ maintainers:
 name: hedera-mirror-rest
 sources:
   - https://github.com/hashgraph/hedera-mirror-node
-version: 0.19.0-rc1
+version: 0.19.0-rc2

--- a/charts/hedera-mirror/Chart.lock
+++ b/charts/hedera-mirror/Chart.lock
@@ -1,13 +1,13 @@
 dependencies:
 - name: hedera-mirror-grpc
   repository: file://../hedera-mirror-grpc
-  version: 0.19.0-rc1
+  version: 0.19.0-rc2
 - name: hedera-mirror-importer
   repository: file://../hedera-mirror-importer
-  version: 0.19.0-rc1
+  version: 0.19.0-rc2
 - name: hedera-mirror-monitor
   repository: file://../hedera-mirror-monitor
-  version: 0.19.0-rc1
+  version: 0.19.0-rc2
 - name: postgresql-ha
   repository: https://charts.bitnami.com/bitnami
   version: 6.8.4
@@ -16,9 +16,9 @@ dependencies:
   version: 12.10.1
 - name: hedera-mirror-rest
   repository: file://../hedera-mirror-rest
-  version: 0.19.0-rc1
+  version: 0.19.0-rc2
 - name: timescaledb-single
   repository: https://raw.githubusercontent.com/timescale/timescaledb-kubernetes/master/charts/repo/
   version: 0.8.2
-digest: sha256:f5d7591e319907ccc6b8cf1d7ee79988967195baa5876e4a6afca29561ac471c
-generated: "2021-04-20T16:31:37.600539-05:00"
+digest: sha256:ec30353362989b2805146a1f123823e7d0eb37311e80f8b65ef14e679a7b4fa5
+generated: "2021-04-26T19:52:40.164649-05:00"

--- a/charts/hedera-mirror/Chart.yaml
+++ b/charts/hedera-mirror/Chart.yaml
@@ -1,5 +1,5 @@
 apiVersion: v2
-appVersion: "0.32.0-rc1"
+appVersion: "0.32.0-rc2"
 dependencies:
   - alias: grpc
     condition: grpc.enabled
@@ -57,4 +57,4 @@ maintainers:
 name: hedera-mirror
 sources:
   - https://github.com/hashgraph/hedera-mirror-node
-version: 0.19.0-rc1
+version: 0.19.0-rc2

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -17,7 +17,7 @@ services:
       - 5432:5432
 
   grpc:
-    image: gcr.io/mirrornode/hedera-mirror-grpc:0.32.0-rc1
+    image: gcr.io/mirrornode/hedera-mirror-grpc:0.32.0-rc2
     restart: unless-stopped
     environment:
       HEDERA_MIRROR_GRPC_DB_HOST: db
@@ -29,7 +29,7 @@ services:
       - 5600:5600
 
   importer:
-    image: gcr.io/mirrornode/hedera-mirror-importer:0.32.0-rc1
+    image: gcr.io/mirrornode/hedera-mirror-importer:0.32.0-rc2
     restart: unless-stopped
     environment:
       HEDERA_MIRROR_IMPORTER_DB_HOST: db
@@ -41,7 +41,7 @@ services:
   monitor:
     deploy:
       replicas: 0
-    image: gcr.io/mirrornode/hedera-mirror-monitor:0.32.0-rc1
+    image: gcr.io/mirrornode/hedera-mirror-monitor:0.32.0-rc2
     restart: unless-stopped
     environment:
       SPRING_CONFIG_ADDITIONAL_LOCATION: file:/usr/etc/hedera-mirror-monitor/
@@ -58,7 +58,7 @@ services:
       - 6379:6379
 
   rest:
-    image: gcr.io/mirrornode/hedera-mirror-rest:0.32.0-rc1
+    image: gcr.io/mirrornode/hedera-mirror-rest:0.32.0-rc2
     environment:
       HEDERA_MIRROR_REST_DB_HOST: db
     restart: unless-stopped
@@ -67,7 +67,7 @@ services:
       - 5551:5551
 
   rosetta:
-    image: gcr.io/mirrornode/hedera-mirror-rosetta:0.32.0-rc1
+    image: gcr.io/mirrornode/hedera-mirror-rosetta:0.32.0-rc2
     environment:
       HEDERA_MIRROR_ROSETTA_DB_HOST: db
     restart: unless-stopped

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/PgCopy.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/PgCopy.java
@@ -34,6 +34,7 @@ import java.util.Collection;
 import java.util.stream.Collectors;
 import lombok.extern.log4j.Log4j2;
 import org.postgresql.PGConnection;
+import org.postgresql.copy.CopyIn;
 import org.postgresql.copy.PGCopyOutputStream;
 
 import com.hedera.mirror.importer.converter.ByteArrayToHexSerializer;
@@ -82,9 +83,14 @@ public class PgCopy<T> {
         try {
             Stopwatch stopwatch = Stopwatch.createStarted();
             PGConnection pgConnection = connection.unwrap(PGConnection.class);
+            CopyIn copyIn = pgConnection.getCopyAPI().copyIn(sql);
 
-            try (var pgCopyOutputStream = new PGCopyOutputStream(pgConnection, sql, properties.getBufferSize())) {
+            try (var pgCopyOutputStream = new PGCopyOutputStream(copyIn, properties.getBufferSize())) {
                 writer.writeValue(pgCopyOutputStream, items);
+            } finally {
+                if (copyIn.isActive()) {
+                    copyIn.cancelCopy();
+                }
             }
 
             insertDurationMetric.record(stopwatch.elapsed());

--- a/hedera-mirror-rest/api/v1/openapi.yml
+++ b/hedera-mirror-rest/api/v1/openapi.yml
@@ -550,7 +550,7 @@ tags:
     description: The tokens object represents the information associated with a token entity and returns a list of token information.
 info:
   title: Hedera Mirror Node REST API
-  version: 0.32.0-rc1
+  version: 0.32.0-rc2
   license:
     name: Apache-2.0
     url: 'https://www.apache.org/licenses/LICENSE-2.0.html'

--- a/hedera-mirror-rest/monitoring/monitor_apis/package.json
+++ b/hedera-mirror-rest/monitoring/monitor_apis/package.json
@@ -1,6 +1,6 @@
 {
   "name": "hedera-mirror-monitor",
-  "version": "0.32.0-rc1",
+  "version": "0.32.0-rc2",
   "description": "Hedera Mirror Node Monitor",
   "main": "server.js",
   "private": true,

--- a/hedera-mirror-rest/package.json
+++ b/hedera-mirror-rest/package.json
@@ -1,6 +1,6 @@
 {
   "name": "hedera-mirror-rest",
-  "version": "0.32.0-rc1",
+  "version": "0.32.0-rc2",
   "description": "Hedera Mirror Node REST API",
   "main": "server.js",
   "private": true,

--- a/hedera-mirror-rosetta/config/application.yml
+++ b/hedera-mirror-rosetta/config/application.yml
@@ -15,4 +15,4 @@ hedera:
       port: 5700
       realm: 0
       shard: 0
-      version: 0.32.0-rc1
+      version: 0.32.0-rc2

--- a/hedera-mirror-test/src/test/resources/features/schedule/schedule.feature
+++ b/hedera-mirror-test/src/test/resources/features/schedule/schedule.feature
@@ -19,7 +19,6 @@ Feature: Schedule Base Coverage Feature
             | accountName | httpStatusCode |
             | "CAROL"     | 200            |
 
-    @acceptance @Acceptance
     Scenario Outline: Validate Base Schedule Flow - ScheduleCreate of CryptoAccountCreate and ScheduleDelete
         Given I successfully schedule a crypto account create
         When the network confirms schedule presence

--- a/hedera-mirror-test/src/test/resources/k8s/hcs-perf-message-submit.yml
+++ b/hedera-mirror-test/src/test/resources/k8s/hcs-perf-message-submit.yml
@@ -41,7 +41,7 @@ spec:
                     app.kubernetes.io/name: test
       restartPolicy: Never
       containers:
-        - image: gcr.io/mirrornode/hedera-mirror-test:0.32.0-rc1
+        - image: gcr.io/mirrornode/hedera-mirror-test:0.32.0-rc2
           name: test
           env:
             - name: testProfile

--- a/hedera-mirror-test/src/test/resources/k8s/hcs-perf-publish-test.yml
+++ b/hedera-mirror-test/src/test/resources/k8s/hcs-perf-publish-test.yml
@@ -43,7 +43,7 @@ spec:
                     app.kubernetes.io/name: test
       restartPolicy: Never
       containers:
-        - image: gcr.io/mirrornode/hedera-mirror-test:0.32.0-rc1
+        - image: gcr.io/mirrornode/hedera-mirror-test:0.32.0-rc2
           name: test
           env:
             - name: testProfile

--- a/hedera-mirror-test/src/test/resources/k8s/hcs-perf-subscribe-test.yml
+++ b/hedera-mirror-test/src/test/resources/k8s/hcs-perf-subscribe-test.yml
@@ -46,7 +46,7 @@ spec:
                     app.kubernetes.io/name: test
       restartPolicy: Never
       containers:
-        - image: gcr.io/mirrornode/hedera-mirror-test:0.32.0-rc1
+        - image: gcr.io/mirrornode/hedera-mirror-test:0.32.0-rc2
           name: test
           env:
             - name: testProfile

--- a/hedera-mirror-test/src/test/resources/k8s/hts-perf-batch-publish-batch-validate.yml
+++ b/hedera-mirror-test/src/test/resources/k8s/hts-perf-batch-publish-batch-validate.yml
@@ -47,7 +47,7 @@ spec:
                     app.kubernetes.io/name: test
       restartPolicy: Never
       containers:
-        - image: gcr.io/mirrornode/hedera-mirror-test:0.32.0-rc1
+        - image: gcr.io/mirrornode/hedera-mirror-test:0.32.0-rc2
           name: test
           env:
             - name: testProfile

--- a/hedera-mirror-test/src/test/resources/k8s/hts-perf-publish-and-retrieve.yml
+++ b/hedera-mirror-test/src/test/resources/k8s/hts-perf-publish-and-retrieve.yml
@@ -46,7 +46,7 @@ spec:
                     app.kubernetes.io/name: test
       restartPolicy: Never
       containers:
-        - image: gcr.io/mirrornode/hedera-mirror-test:0.32.0-rc1
+        - image: gcr.io/mirrornode/hedera-mirror-test:0.32.0-rc2
           name: test
           env:
             - name: testProfile

--- a/pom.xml
+++ b/pom.xml
@@ -75,8 +75,8 @@
         <micrometer-jvm-extras.version>0.2.2</micrometer-jvm-extras.version>
         <msgpack.version>0.8.22</msgpack.version>
         <protobuf.version>3.15.8</protobuf.version>
-        <release.version>0.32.0-rc1</release.version> <!-- Used to replace release versions in all files -->
-        <release.chartVersion>0.19.0-rc1</release.chartVersion>
+        <release.version>0.32.0-rc2</release.version> <!-- Used to replace release versions in all files -->
+        <release.chartVersion>0.19.0-rc2</release.chartVersion>
         <replacer.version>1.5.3</replacer.version>
         <!--suppress UnresolvedMavenProperty -->
         <sonar.host.url>https://sonarcloud.io</sonar.host.url>

